### PR TITLE
chore(ci) omit tests from license scan

### PR DESCRIPTION
**Pull request by rainest**
**What this PR does / why we need it**:
Removes the `test` directory before initiating the license scan, to omit test-only dependencies from license reports.

We include our application and its tests in the same repo. The tests aren't compiled into any release artifacts, so it doesn't make sense to include them in the license report we ship with those artifacts. They're also the source of a fair number of license issue reports that we need to investigate and ignore.

Per FOSSA, there's no Go-specific functionality to ignore certain build tags, but deleting those files has the same effect, and we conveniently have those under the same directory (not units, but they're not usually the source of spurious reports because they usually don't add dependencies their package didn't already have).

See [test.txt](https://github.com/Kong/kubernetes-ingress-controller/files/9245901/test.txt) for a POC example with some extra `ls` calls thrown into the job.

### CODE REVIEW

**Review by shaneutt — DISMISSED** [review]

File: .github/workflows/license.yaml [hunk 1]
```diff
@@ -10,6 +10,8 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
+      - shell: bash
```
Comment (shaneutt) [inline .github/workflows/license.yaml:13]: I realize, we should add a comment around this to explain.

**Review by mflendrich — CHANGES_REQUESTED** [review]
I am not convinced that "not included in the artifact we ship" is a strong enough premise to imply that it would render a license we include not worth looking at.

Hypothetically there might be licenses that have unexpected implications even if not compiled directly into the artifact, but just included in the repo. I don't have an immediate answer whether we're likely to come across/include something with such a license, though.

I don't think that we should be making this change.

**Comment by rainest** [comment]
Added `go mod tidy` as well and had to rearrange some stuff outside of test that linked to test (it was a package that was only used by stuff under `/test`, I think we had it elsewhere because we didn't have an `internal` directory under `test/` earlier).

This breaks in act for some reason, so https://github.com/rainest/kubernetes-ingress-controller/runs/7657773046?check_suite_focus=true

**Comment by rainest** [comment]
Closed pending feedback from legal.

**Pull request closed without merging**